ddl/job_table.go

```diff
@@ -220,6 +223,108 @@ func (d *ddl) delivery2worker(wk *worker, pool *workerPool, job *model.Job) {
 	})
 }
 
+// getDDLReorgHandle get ddl reorg handle.
+func getDDLReorgHandle(sess *session, job *model.Job) (element *meta.Element, startKey, endKey kv.Key, physicalTableID int64, err error) {
+	sql := fmt.Sprintf("select ele_id, ele_type, start_key, end_key, physical_id from mysql.tidb_ddl_reorg where job_id = %d", job.ID)
+	rows, err := sess.execute(context.Background(), sql, "get_handle")
+	if err != nil {
+		return nil, nil, nil, 0, err
+	}
+	if len(rows) == 0 {
+		return nil, nil, nil, 0, meta.ErrDDLReorgElementNotExist
+	}
+	id := rows[0].GetInt64(0)
+	tp := rows[0].GetBytes(1)
+	element = &meta.Element{
+		ID:      id,
+		TypeKey: tp,
+	}
+	startKey = rows[0].GetBytes(2)
+	endKey = rows[0].GetBytes(3)
+	physicalTableID = rows[0].GetInt64(4)
+	// physicalTableID may be 0, because older version TiDB (without table partition) doesn't store them.
+	// update them to table's in this case.
+	if physicalTableID == 0 {
```

Comment:
Can we get rid of this? `physicalTableID == 0` only exists in TiDB 2.x.

Reply:
this PR we should just keep the same as before, removing this will be another story